messaging-core/src/main/java/org/fenixedu/messaging/core/domain/Message.java

```diff
@@ -191,95 +192,82 @@ public MessageBuilder content(LocalizedString subject, LocalizedString textBody,
             return this;
         }
 
-        private static <T> void filteredAdd(Stream<T> items, Collection<T> collection) {
-            items.filter(Objects::nonNull).forEach(collection::add);
+        private static <T> void filteredAdd(Stream<T> stream, Predicate<T> filter, Set<T> set) {
+            if(stream != null) {
```

Comment:
The thing is: if everywhere else you return an empty stream instead of null, there would be no null checks anywhere